### PR TITLE
USWDS - Analytics: Add GA4 analytics tags

### LINKED ISSUE #2390: USWDS-Site - Feature: Add GA4 analytics snippet
### Is your feature request related to a problem? Please describe.

We'll need to add a new code snippet to our websites so we can transition to GA4 data collection under the GSA Enterprise Analytics account while continuing to collect UA data on our existing account.

### Describe the solution you'd like

The existing DAP code in `_includes/analytics.html` should continue to appear within the document `head`, but this new GA4 snippet AND our existing UA snippet should now appear directly _after_ the `head`.

[DAP implementation guide](https://github.com/digital-analytics-program/gov-wide-code/blob/master/documentation/GSA%20DAP%204.1%20-%20Quick%20Guide.pdf) (PDF) → 
[GA4 Implementation guide](https://support.google.com/analytics/answer/9304153?sjid=7848239909584944126-NC#zippy=%2Cadd-the-google-tag-directly-to-your-web-pages) (see "Set up data collection for websites") → 

In the `head`:
```html
<--! DAP -->
<script async type="text/javascript" id="_fed_an_ua_tag" src="https://dap.digitalgov.gov/Universal-Federated-Analytics-Min.js?agency=GSA&subagency=TTS"></script>
```

Directly after `head`:
```html
<--! GA4 -->
<!-- Google tag (gtag.js) --> <script async src="https://www.googletagmanager.com/gtag/js?id=G-HBYXWFP794"></script> <script> window.dataLayer = window.dataLayer || []; function gtag(){dataLayer.push(arguments);} gtag('js', new Date()); gtag('config', 'G-HBYXWFP794'); </script>

<--! UA -->
<script>
	window.ga=window.ga||function(){(ga.q=ga.q||[]).push(arguments)};ga.l=+new Date;
	ga('create', '{{ site.google_analytics_ua }}', 'gsa.gov');
	ga('send', 'pageview');
	// anonymize user IPs (chops off the last IP triplet)
	ga('set', 'anonymizeIp', true);

	// forces SSL even if the page were somehow loaded over http://
	ga('set', 'forceSSL', true);

	ga('send', 'pageview');
</script>
<script async src='https://www.google-analytics.com/analytics.js'></script>
```

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://designsystem.digital.gov/about/community/#community-conduct).
- [X] I checked the [current issues](https://github.com/uswds/uswds-site/issues) for duplicate feature requests.

### PR DESCRIPTION
**Pull request by mahoneycm**
# Summary

Integrates GA4 analytics to Site.

## Related issue

Closes #2390

## Preview link

[Preview link →](https://federalist-ead78f8d-8948-417c-a957-c21ec5617a57.sites.pages.cloud.gov/preview/uswds/uswds-site/cm-GA4-analytics/)

## Problem statement

We'll need to add a new code snippet to our websites so we can transition to GA4 data collection under the GSA Enterprise Analytics account while continuing to collect UA data on our existing account.

## Solution

Add GA4 analytics scripts to our site.

> [!TIP]
Guidance about where to add the tag was a bit confusing. When it says after `<head>` I believe it means after the *opening* tag. Adding scripts after the closing tag does not properly add the scripts when inspecting the site and they do not appear to be loaded in the network tab.
>
> This also appears to match where the analytics tag can be found on instructions page when inspecting the page.

## Major changes

- Analytics partial now includes GA4 tag
- Moved analytics directly under opening `<head>` tag
    - We can revert this if we prefer to have meta data at the top of our head

## Testing and review

1. Visit the site preview and open devTools.
2. Navigate to the network tab and refresh the page.
3. You should see three analytics scripts loaded directly after `localhost`.
4. Their respective Request URL for each should be:
    - [ ]  `www.googletagmanager.com`
    - [ ]  `www.google-analytics.com`
    - [ ]  `dap.digitalgov.gov`

### Checking within Google Analytics

I'm curious if there is any information available to us within the Google Analytics manager. Do the federalist previews that contain the analytics tag send data we could check?

### CODE REVIEW

File: _includes/analytics.html [hunk 1]
```diff
@@ -1,5 +1,9 @@
-	<!-- Google Analytics -->
-
+<!-- Google Analytics -->
+	<!-- GA4 -->
+	<!-- Google tag (gtag.js) --> <script async src="https://www.googletagmanager.com/gtag/js?id=G-HBYXWFP794"></script> 
+	<script> window.dataLayer = window.dataLayer || []; function gtag(){dataLayer.push(arguments);} gtag('js', new Date()); gtag('config', 'G-HBYXWFP794'); </script>
```
Comment (mejiaj) [inline _includes/analytics.html:4]: ## Optional

We can condense some comments & move the `<script>` to its own line.

```suggestion
<!-- Google Analytics - GA4 -->
	<!-- Google tag (gtag.js) --> 
	<script async src="https://www.googletagmanager.com/gtag/js?id=G-HBYXWFP794"></script> 
	<script> window.dataLayer = window.dataLayer || []; function gtag(){dataLayer.push(arguments);} gtag('js', new Date()); gtag('config', 'G-HBYXWFP794'); </script>
```

File: _includes/analytics.html [hunk 2]
```diff
@@ -1,5 +1,9 @@
...
+	
+	<!-- UA -->
 	<script>
 		window.ga=window.ga||function(){(ga.q=ga.q||[]).push(arguments)};ga.l=+new Date;
 		ga('create', '{{ site.google_analytics_ua }}', 'gsa.gov');
```
Comment (mejiaj) [inline _includes/analytics.html:9]: No issues with keeping the old code, but some questions.

1. Will this conflict with [GA4 guideance](https://support.google.com/analytics/answer/9744165?hl=en#zippy=%2Cin-this-article:~:text=You%20will%20see%20some%20of%20the%20following%20options%2C%20depending%20on%20how%20the%20site%20is%20configured.)?
2. UA is deprecated, should we remove lines `6-9`?
3. How can we ensure the new code is working?

**Review by mejiaj — COMMENTED** [review]
@mahoneycm, looks good. Just a few questions to better understand this implementation.

File: _includes/analytics.html (hunk 1 above)
Comment (mahoneycm) [reply _includes/analytics.html:4]: LGTM!

File: _includes/analytics.html (hunk 2 above)
Comment (mahoneycm) [reply _includes/analytics.html:9]: 1. Based on the guidance, the new GA4 property works to collect data alongside the existing UA property, which remains unchanged. Original instructions from Dan in #2390 mentioned keeping the UA tag. That said...
2. UA does appear to no longer be collecting new data as of July 1, 2023. I haven't seen a definitive, "Yes, this is no longer needed" but as long as we don't need this tag here to _access_ the API or it's data, then we should be good to remove.
3. There are two things we'll need to check to make sure the code is working
  a. The scrips should be loading and recognized in the network tab _(See testing instructions)_
  b. We'll need Dan to log into https://analytics.google.com/ and either finish any additional setup or confirm that the data is received.

**Review by thisisdano — APPROVED** [review]
I agree with the approach here and sorry for the confusion!

File: _includes/analytics.html (hunk 2 above)
Comment (annepetersen) [reply _includes/analytics.html:9]: Re: 2., I logged in and as far as I can see, new data is still being collected, FWIW.

**Comment by thisisdano** [comment]
Failing `htmlproofer` — can you update with the most recent version of `main`?

**Comment by mejiaj** [comment]
@thisisdano @mahoneycm I've merged in the latest from `main`.

**Comment by thisisdano** [comment]
merging over pa11y crash

**Pull request merged**